web/src/utils/datetime.ts

```diff
@@ -44,3 +45,30 @@ export const getTimeRangeOptions = (t: TFunction) => {
 export const getFormattedDate = (date: Date, format = 'llll') => {
   return moment(date).format(format);
 };
+
+export const rangeToSeconds = (range: TimeRange | number): number => {
+  if (typeof range === 'number') {
+    return range;
+  }
+  return range.to - range.from;
+};
+
+export const computeStepInterval = (range: TimeRange | number) => {
+  const seconds = rangeToSeconds(range);
+  let interval = Math.floor(seconds / 10);
+  if (interval < 30) {
+    interval = 30;
+  }
+  const step = Math.floor(interval / 2);
+  return {
+    rateIntervalSeconds: interval,
+    stepSeconds: step
+  };
+};
```

Comment:
I'm still having issues with the number of values returned by loki here.
On 1 day range I only get 2 values per metrics

Reply:
I have updated LLC `retention_period` to 72h. Let's see how it goes.

Reply:
I'll double check tomorrow as well

Reply:
Just checked right now: actually, there's no flow before 8:15 this morning :

![image](https://user-images.githubusercontent.com/2153442/194512966-04c843ba-c717-4b69-8675-b2e2cc31c2a7.png)

So that explains why there's so few datapoints. It seems like the change I'm doing is only revealing a problem that pre-existed.

Reply:
So, now flows seem to start at 10:00. It's like there's a 2hrs30min TTL configured somewhere 🤔

Reply:
I restarted the loki pods to ensure configuration is up to date. We should get more at then end of the day and 72h on monday